pkg/scheduler/partition.go

```diff
@@ -922,6 +923,16 @@ func (pc *PartitionContext) GetApplications() []*objects.Application {
 	return appList
 }
 
+func (pc *PartitionContext) GetCompletedApplications() []*objects.Application {
+	pc.RLock()
+	defer pc.RUnlock()
+	var appList []*objects.Application
+	for _, app := range pc.completedApplications {
+		appList = append(appList, app)
```

Comment:
this needs to become a filtering loop as we only have one map: we can directly filter on the `Expired` state if I am correct